### PR TITLE
doc: Adjusting `advanced_cluster` documentation reflecting changes in how auto scaling properties work

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

Link to any related issue(s): CLOUDP-286228

Follow up PR will be done to adjust [new sharding configuration migration guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/advanced-cluster-new-sharding-schema) with consideration for auto scaling.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: docs/data-sources/advanced_cluster.md [hunk 1]
```diff
@@ -136,7 +136,7 @@ Key-value pairs that categorize the cluster. Each key and value has a maximum le
 
 ### replication_specs
 
-* `id` - **(DEPRECATED)** Unique identifer of the replication document for a zone in a Global Cluster. This value corresponds to the legacy sharding schema (no independent shard scaling) and is different from the Shard ID you may see in the Atlas UI.
+* `id` - **(DEPRECATED)** Unique identifer of the replication document for a zone in a Global Cluster. This value corresponds to the legacy sharding schema (no independent shard scaling) and is different from the Shard ID you may see in the Atlas UI. This value is not populated when a sharded cluster has independent scaled shards.
```
Comment (AgustinBettati) [inline docs/data-sources/advanced_cluster.md]: this is a pass by clarification of existing behaviour. This is currently mentioned in our [migration guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/advanced-cluster-new-sharding-schema) but not here.

File: docs/resources/advanced_cluster.md [hunk 2]
```diff
@@ -596,55 +596,45 @@ If you are upgrading a replica set to a sharded cluster, you cannot increase the
...
-    - Set to `true` to enable disk auto-scaling.
-    - Set to `false` to disable disk auto-scaling.
 
-~> **IMPORTANT:** If `disk_gb_enabled` is true, Atlas automatically scales the cluster up or down.
-This will cause the value of `replication_specs.#.region_config.#.(analytics_specs|electable_specs|read_only_specs).disk_size_gb` returned to potentially be different than what is specified in the Terraform config and if you use, and then apply, not noting this, Terraform will scale the cluster disk size back to the original `disk_size_gb` value.
-To prevent disk scaling use a lifecycle customization:
-`lifecycle {
-  ignore_changes = [replication_specs.#.region_config.#.electable_specs.disk_size_gb]
-}`
-After adding the `lifecycle` block to explicitly change `replication_specs.#.region_config.#.(analytics_specs|electable_specs|read_only_specs).disk_size_gb` comment out the `lifecycle` block and run `terraform apply`. Please be sure to uncomment the `lifecycle` block once done to prevent any accidental changes.
```
Comment (AgustinBettati) [inline docs/resources/advanced_cluster.md:608]: Unified this important note into a single one for both `disk_gb_enabled` and `compute_enabled`. The concept of values which can change and how to ignore them is the same for both cases.

File: docs/resources/advanced_cluster.md [hunk 3]
```diff
@@ -596,55 +596,45 @@ If you are upgrading a replica set to a sharded cluster, you cannot increase the
...
+* `compute_enabled` - (Optional) Flag that indicates whether instance size auto-scaling is enabled. This parameter defaults to false. If a sharded cluster is making use of the [New Sharding Configuration](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/advanced-cluster-new-sharding-schema), auto-scaling of the instance size will be independent for each individual shard. On the contrary, if a sharded cluster makes use of deprecated `num_shards` attribute (with values > 1), instance size auto-scaling will be performed uniformily across all shards in the cluster.
```
Comment (AgustinBettati) [inline docs/resources/advanced_cluster.md:600]: relevant mention of new auto-scaling behaviour is here

**Review by lantoli — APPROVED** [review]

File: docs/data-sources/advanced_cluster.md (hunk 1 above)
Comment (jwilliams-mongo) [reply docs/data-sources/advanced_cluster.md]: question: is this null or an empty string when using independent shard scaling? Not sure if there's a material difference, but good to clarify

File: docs/data-sources/advanced_clusters.md [hunk 4]
```diff
@@ -138,7 +138,7 @@ Key-value pairs that categorize the cluster. Each key and value has a maximum le
 
 ### replication_specs
 
-* `id` - **(DEPRECATED)** Unique identifer of the replication document for a zone in a Global Cluster. This value corresponds to the legacy sharding schema (no independent shard scaling) and is different from the Shard ID you may see in the Atlas UI.
+* `id` - **(DEPRECATED)** Unique identifer of the replication document for a zone in a Global Cluster. This value corresponds to the legacy sharding schema (no independent shard scaling) and is different from the Shard ID you may see in the Atlas UI. This value is not populated when a sharded cluster has independent scaled shards.
```
Comment (jwilliams-mongo) [inline docs/data-sources/advanced_clusters.md]: same comment as above

File: docs/resources/advanced_cluster.md [hunk 5]
```diff
@@ -533,7 +533,7 @@ replication_specs {
 }
 ```
 
-* `id` - **(DEPRECATED)** Unique identifer of the replication document for a zone in a Global Cluster. This value corresponds to the legacy sharding schema (no independent shard scaling) and is different from the Shard ID you may see in the Atlas UI.
+* `id` - **(DEPRECATED)** Unique identifer of the replication document for a zone in a Global Cluster. This value corresponds to the legacy sharding schema (no independent shard scaling) and is different from the Shard ID you may see in the Atlas UI. This value is not populated when a sharded cluster has independent scaled shards.
```
Comment (jwilliams-mongo) [inline docs/resources/advanced_cluster.md]: same comment as above

**Review by jwilliams-mongo — APPROVED** [review]
LGTM % a few minor comments

File: docs/data-sources/advanced_cluster.md (hunk 1 above)
Comment (jwilliams-mongo) [inline docs/data-sources/advanced_cluster.md]:
```suggestion
* `id` - **(DEPRECATED)** Unique identifer of the replication document for a zone in a Global Cluster. This value corresponds to the legacy sharding schema (no independent shard scaling) and is different from the Shard ID you may see in the Atlas UI. This value is not populated when a sharded cluster has independently scaled shards.
```

Comment (AgustinBettati) [reply docs/data-sources/advanced_cluster.md]: empty string, made a slight change so that it is clarified 👍

**Pull request merged**